In [18]:
library(repr)
library(tidyverse)
library(ggplot2)
options(repr.matrix.max.rows = 6)

## Individual Planning Report

GitHub repository: https://github.com/wandereryiu/zhao-yiyu-project-report-group38

In [15]:
players_data <- read_csv("data/players.csv")

sessions_data <- read_csv("data/sessions.csv") # Not needed for current report

players_data

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


experience subscribe
1   Pro         TRUE    
2   Veteran     TRUE    
3   Veteran    FALSE    
⋮   ⋮          ⋮        
194 Amateur    FALSE    
195 Amateur    FALSE    
196 Pro         TRUE    
    hashedEmail                                                     
1   f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
2   f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9
3   b674dd7ee0d24096d1c019615ce4d12b20fcbff12d79d3c5a9d2118eb7ccbb28
⋮   ⋮                                                               
194 d572f391d452b76ea2d7e5e53a3d38bfd7499c7399db299bd4fedb06a46ad5bb
195 f19e136ddde68f365afc860c725ccff54307dedd13968e896a9f890c40aea436
196 d9473710057f7d42f36570f0be83817a4eea614029ff90cf50d8889cdd729d11
    played_hours name      gender            Age
1   30.3         Morgan    Male               9 
2    3.8         Christian Male              17 
3    0.0         Blake     Male              17 
⋮   ⋮            ⋮         ⋮                 ⋮  
194 0.0          Dylan     Prefer not to say 57 
195 2.3          Harlow    Male              17 
196 0.2          Ahmed     Other             NA

### Data Description

#### Issues with Data

An issue that I notice with the data set given is that there are observations of players with 0.0 hours played hours, which I assume means that they did not contribute data. In this case, it may be better to exclude these observations from the data when computing summary statistics.

There are also some observations in which players do not have an age, which could also affect the data. These observations may also be excluded.



In [37]:
# Filter out unwanted data

players_filtered <- players_data |>
    filter(played_hours != 0.0, 
           !is.na(Age))

players_filtered

experience subscribe
1   Pro        TRUE     
2   Veteran    TRUE     
3   Amateur    TRUE     
⋮   ⋮          ⋮        
107 Veteran     TRUE    
108 Veteran    FALSE    
109 Amateur    FALSE    
    hashedEmail                                                     
1   f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
2   f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9
3   23fe711e0e3b77f1da7aa221ab1192afe21648d47d2b4fa7a5a659ff443a0eb5
⋮   ⋮                                                               
107 ba24bebe588a34ac546f8559850c65bc90cd9d51b821581bd6e25cff437a1081
108 71453e425f07d10da4fa2b349c83e73ccdf0fb3312f778b35c5802c3292c87bd
109 f19e136ddde68f365afc860c725ccff54307dedd13968e896a9f890c40aea436
    played_hours name      gender Age
1   30.3         Morgan    Male    9 
2    3.8         Christian Male   17 
3    0.7         Flora     Female 21 
⋮   ⋮            ⋮         ⋮      ⋮  
107 0.1          Gabriela  Female 44 
108 0.3          Pascal    Male   22 
109 2.3          Harlow    Male   17

#### Data Description

The original data set of players.csv had 196 observations. After removing observations in which the played hours was 0.0 and the age was NA, there were 109 observations remaining. 



### Data Analysis and Visualization

The question I will be be addressing for this report is:

**1. What player characteristics and behaviours are most predictive of subscribing to a game-related newsletter, and how do these features differ between various player types?**